packages/edit-post/src/components/layout/index.js

```diff
@@ -151,6 +156,117 @@ function useEditorStyles() {
 	] );
 }
 
+/**
+ * @param {Object}  props
+ * @param {boolean} props.isLegacy True when the editor canvas is not in an iframe.
+ */
+function MetaBoxesMain( { isLegacy } ) {
+	const [ isOpen, openHeight, hasAnyVisible ] = useSelect( ( select ) => {
+		const { get } = select( preferencesStore );
+		const { isMetaBoxLocationVisible } = select( editPostStore );
+		return [
+			get( 'core/edit-post', 'metaBoxesMainIsOpen' ),
+			get( 'core/edit-post', 'metaBoxesMainOpenHeight' ),
+			isMetaBoxLocationVisible( 'normal' ) ||
+				isMetaBoxLocationVisible( 'advanced' ),
+		];
+	}, [] );
+	const { set: setPreference } = useDispatch( preferencesStore );
+	const resizableBoxRef = useRef();
+	const isShort = useMediaQuery( '(max-height: 549px)' );
+
+	const isAutoHeight = openHeight === undefined;
+	// In case a user size is set stops the default max-height from applying.
+	useLayoutEffect( () => {
+		if ( ! isLegacy && hasAnyVisible && ! isShort && ! isAutoHeight ) {
+			resizableBoxRef.current.resizable.classList.add( 'has-user-size' );
+		}
+	}, [ isAutoHeight, isShort, hasAnyVisible, isLegacy ] );
+
+	if ( ! hasAnyVisible ) {
+		return;
+	}
+
+	const className = 'edit-post-meta-boxes-main';
+	const contents = (
+		<div
+			className={ clsx(
+				// The class name 'edit-post-layout__metaboxes' is retained because some plugins use it.
+				'edit-post-layout__metaboxes',
+				! isLegacy && 'edit-post-meta-boxes-main__liner'
+			) }
+		>
+			<MetaBoxes location="normal" />
+			<MetaBoxes location="advanced" />
+		</div>
+	);
+
+	if ( isLegacy ) {
+		return contents;
+	}
+
+	if ( isShort ) {
+		return (
+			<details
+				className={ className }
+				open={ isOpen }
+				onToggle={ ( { target } ) => {
+					setPreference(
+						'core/edit-post',
+						'metaBoxesMainIsOpen',
+						target.open
+					);
+				} }
+			>
+				<summary>{ __( 'Meta Boxes' ) }</summary>
+				{ contents }
+			</details>
+		);
+	}
+	return (
+		<ResizableBox
+			className={ className }
+			defaultSize={ { height: openHeight } }
+			ref={ resizableBoxRef }
+			enable={ {
+				top: true,
+				right: false,
+				bottom: false,
+				left: false,
+				topLeft: false,
+				topRight: false,
+				bottomRight: false,
+				bottomLeft: false,
+			} }
+			// This is overriden by an !important rule that applies until user resizes.
+			maxHeight="100%"
```

Comment:
Nit: The same style is defined [here](https://github.com/WordPress/gutenberg/pull/64351/files#diff-e0ee63e1380dc04eba95c99d0d5587304ec37fbaa50fb030beb0ba75576f27fbR8), so is this prop unnecessary?

Reply:
Actually, a brief retest, neither the prop nor the rule in CSS seem necessary 😅. The branch I have that adds keyboard support will use this prop with specified pixel value (for reasons which which I’ll elaborate in its PR) and I’ll leave this be here. On the other maybe we can remember to check if the CSS rule has any purpose.